org.abapgit.adt.backend/src/org/abapgit/adt/backend/internal/RepositoryService.java

```diff
@@ -256,4 +259,75 @@ public URI getURIFromAtomLink(IRepository repository, String relation) {
 		return null;
 	}
 
+	@Override
+	public IAbapGitPullModifiedObjects getModifiedObjects(IProgressMonitor monitor, IRepository currRepository,
+			String user, String password) {
+		//This URI is not a part of the URI's associated with a repository in the backend.
+		//Do/Should we add this to that list in the backend and then fetch the URI from Atom Link, like we do for other URI
+		String uriToModifiedObjectsString = getURIFromAtomLink(currRepository, IRepositoryService.RELATION_PULL).toString()
+				+ "/modifiedobjects"; //$NON-NLS-1$
+		URI uriToModifiedObjects = URI.create(uriToModifiedObjectsString);
+
+		IHeaders headers = null;
+		if (user != null && password != null) {
+			headers = getHttpHeadersForCredentials(user, password);
+		}
+
+		IRestResource restResource = AdtRestResourceFactory.createRestResourceFactory()
+				.createResourceWithStatelessSession(uriToModifiedObjects,
+				this.destinationId);
+
+		IContentHandler<IAbapGitPullModifiedObjects> responseContentHandlerV1 = new AbapGitPullModifiedObjectsContentHandlerV1();
+		restResource.addContentHandler(responseContentHandlerV1);
+
+		IAdtCompatibleRestResourceFilter compatibilityFilter = AdtCompatibleRestResourceFilterFactory
+				.createFilter(responseContentHandlerV1);
+		restResource.addRequestFilter(compatibilityFilter);
+		restResource.addResponseFilter(compatibilityFilter);
+
+		return restResource.get(monitor, headers, IAbapGitPullModifiedObjects.class);
+
+	}
+
+	@Override
+	public IAbapObjects pullRepository(IRepository existingRepository, String branch, String transportRequest, String user, String password,
```

Comment:
After 2105 backend reaches all customers, we will remove the older `pullRepository` method.
Should we still refactor it as there will be just one method in future.

Reply:
This is something interesting to be discussed with Andre and Sunil. Because sometimes if we remove the old way of handling, it could lead to issues with the backend code in ADT_Backend. 

Meaning if the ADT_Backend repository is not updated with latest changes provided by SAP, and if a user updates only the client, the client might not work with ADT_Backend.

So what should be the stand-in supporting ADT_Backend by the ADT Client!!

Reply:
So in that case keeping both the versions will help in supporting older backends. Although I think, that will lead to crowding with time.
A workaround is to manually install/downgrade to older working clients via zip, to work with ADT_Backend. 
For this case, we'll keep both classes for now and then decide if we need to refactor it or remove the older version.

Reply:
Yep.